### PR TITLE
perplexity : faster Winogrande via batching

### PR DESCRIPTION
**Pull request by ggerganov**
Similar to #5017 but for the Winogrande evaluation. Here the performance gain is bigger since the tasks are smaller and we can fit more in a batch.

I'm not 100% sure about the correctness of this implementation - the final value is different from `master`:

```
# master
Final Winogrande score(1267 tasks): 67.6401 +/- 1.3149

# PR
Final Winogrande score(1267 tasks): 69.1397 +/- 1.2982
```

It's possible that it fixes some existing issue, but it is also possible that it introduces a bug. The softmax optimization is not applied yet - it can be applied in a follow-up PR

### CODE REVIEW

File: examples/perplexity/perplexity.cpp [hunk 1]
```diff
@@ -423,26 +423,31 @@ static results_perplexity perplexity(llama_context * ctx, const gpt_params & par
...
+            n_tokens,
+            batch.token    + i,
+            nullptr,
+            batch.pos      + i,
+            batch.n_seq_id + i,
+            batch.seq_id   + i,
+            batch.logits   + i,
+            0, 0, 0, // unused
+        };
+
```
Comment (ikawrakow) [inline examples/perplexity/perplexity.cpp:440]: The `llama_kv_cache_seq_rm` call is no longer needed here?

**Review by ikawrakow — APPROVED** [review]

File: examples/perplexity/perplexity.cpp (hunk 1 above)
Comment (ggerganov) [reply examples/perplexity/perplexity.cpp:440]: It's not needed because we clear the entire KV cache before each batch:

https://github.com/ggerganov/llama.cpp/blob/9e4ad80cfc53dda3e08c08b67565ab92cb00e4a1/examples/perplexity/perplexity.cpp#L942-L944

In the old implementation, it was reusing tokens from a previous batch, so the `llama_kv_cache_seq_rm` was used to evict the unused ones (i.e. the second sentence).

**Comment by ggerganov** [comment]
The score with Mistral 7B has changed in the "positive" direction as well, closer to the HF leaderboard results:

```
# master
Final Winogrande score(1267 tasks): 73.6385 +/- 1.2383

# PR
Final Winogrande score(1267 tasks): 74.5856 +/- 1.2236
```

**Comment by ikawrakow** [comment]
> The score with Mistral 7B has changed in the "positive" direction as well, closer to the HF leaderboard results:

Yes, I saw this change, but could not understand the reason.
